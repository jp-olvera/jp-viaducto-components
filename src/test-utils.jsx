import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
