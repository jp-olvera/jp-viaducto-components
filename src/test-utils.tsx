import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
