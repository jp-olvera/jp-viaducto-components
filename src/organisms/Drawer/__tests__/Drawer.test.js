/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../../test-utils';
import { Drawer } from '..';

describe('<Drawer/>', () => {
  test('should be visible', () => {
    render(
      <Drawer active>
        <h1>Title</h1>
      </Drawer>,
    );
    expect(screen.getByText('Title')).toBeVisible();
  });
  test('should no be visible', () => {
    render(
      <Drawer>
        <h1>Title</h1>
      </Drawer>,
    );
    expect(screen.queryByText('Title')).toMatchSnapshot();
  });
  test('should click the button', () => {
    render(
      <Drawer>
        <h1>Title</h1>
      </Drawer>,
    );
    const button = screen.queryByRole('button');
    fireEvent.click(button);
    expect(button).toMatchInlineSnapshot(`
      <button
        class="BareButton__StyledButton-sc-7bpup3-0 fpYbxH"
        type="button"
      >
        X
      </button>
    `);
  });
});
