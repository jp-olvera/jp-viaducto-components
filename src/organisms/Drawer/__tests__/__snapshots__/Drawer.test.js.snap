// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> should no be visible 1`] = `
<h1>
  Title
</h1>
`;
